microsoft_atp/plugin.spec.yaml

```
@@ -21,481 +21,441 @@ hub_tags:
+    type: credential_secret_key
     required: true
+    example: xxxxxxxxxxxxxxxxxxxxx
+    order: 2
+  directory_id:
+    title: Directory ID
+    description: Directory (tenant) ID
     type: string
-  host:
-    title: Host URL
-    description: Base URL for endpoints, e.g. https://wdatp-alertexporter-us.securitycenter.windows.com
     required: true
-    type: string
-    default: https://wdatp-alertexporter-us.securitycenter.windows.com
-  resource_url:
-    title: Resource URL
-    description: This will be the server that will return information from the plug-in
-      requests. This will typically be https://graph.windows.net (for alert information)
-      or https://api.securitycenter.windows.com (for machine information and actions).
-    required: false
-    type: string
-    default: https://api.securitycenter.windows.com
-  seconds_ago:
-    title: Seconds Ago
-    description: Seconds before the first run of the trigger to check for alerts.
-      If left blank, the trigger will start scanning from the time the workflow begins.
-    required: false
-    type: integer
-    default: 0
+    example: xxxx-xxxx-xxxx-xxxx-xxx
+    order: 3
+
 actions:
-  get_machine_id_from_alert_id:
-    title: Get Machine ID from Alert
-    description: Retrieve the machine ID related to an alert
+  get_machine_information:
+    title: Get Machine Information
+    description: Get details about a machine with it's ID
     input:
-      alert_id:
+      machine_id:
+        title: Machine ID
+        description: Machine ID
         type: string
-        title: Alert ID
-        description: Alert ID to get a machine ID from
+        example: 2df36d707c1ee5084cef77f3dbfc95db65bc4a73
         required: true
     output:
-      machine_information:
-        title: Machine Information
-        description: The machine ID related to the given Alert ID
-        type: machine_information_type
+      machine:
+        title: Machine
+        description: Machine information
+        type: object
         required: true
+
   get_file_id_from_alert_id:
-    title: Get File ID from Alert
-    description: Retrieve the file ID related to an alert
+    title: Get Files from Alert
+    description: Retrieve a list of file information objects related to an alert
     input:
       alert_id:
         type: string
         title: Alert ID
         description: Alert ID to get files from
+        example: da637293198146839977_2089064327
         required: true
     output:
-      file_information:
+      file_list:
         title: File Information
         description: The file ID related to the given alert ID
-        type: file_information
+        type: "[]file"
         required: true
   isolate_machine:
     title: Isolate Machine
-    description: Isolate a machine from the network, but keep the connection to Windows
-      ATP open
+    description: Isolate a machine from the network, but keep the connection to Windows ATP open
```

Do we need the last part? I feel like it's confusing. I'm assuming that's just the behavior of ATP

```suggestion
    description: Isolate a machine from the network
```